src/metabase/db/custom_migrations.clj

```diff
@@ -1033,3 +1033,28 @@
 (define-reversible-migration UnifyTimeColumnsType
   (unify-time-column-type! :up)
   (unify-time-column-type! :down))
+
+(define-reversible-migration CardRevisionAddType
+  (let [migrate! (fn [revision]
+                   (let [object    (json/parse-string (:object revision) keyword)
+                         new-object (-> object
+                                        (assoc :type (if (:dataset object)
+                                                       "model"
+                                                       "question")))]
+                     (t2/query {:update :revision
+                                :set    {:object (json/generate-string new-object)}
+                                :where  [:= :id (:id revision)]})))]
+    (run! migrate! (t2/reducible-query {:select [:*]
+                                        :from   [:revision]
+                                        :where  [:= :model "Card"]})))
+  (let [rollback! (fn [revision]
+                    (let [object    (json/parse-string (:object revision) keyword)
+                          new-object (-> object
+                                         (assoc :dataset (= (:type object) "model"))
```

Comment:
Devils advocate: doesn't this branch keeps the `dataset` field maintained? Why do we need to repopulate in this rollback? I think we could save this for rolling back a migration that deleted the column